src/libsyntax/parse/parser.rs

```diff
@@ -6024,7 +6027,13 @@ impl<'a> Parser<'a> {
 
         // Allow a leading :: because the paths are absolute either way.
         // This occurs with "use $crate::..." in macros.
-        self.eat(&token::ModSep);
+        let is_global = self.eat(&token::ModSep);
+
+        // Checking that there is no `self` or `super` in global context
+        if is_global && self.token.is_special_ident_in_modname() {
+            return Err(self.fatal(&format!("expected identifier, found keyword `{}`",
+                                       self.this_token_to_string())));
```

Comment:
nit: this line should start at the same column as the third `(` in the above line